### PR TITLE
feat: add handwritten serialize

### PR DESCRIPTION
**Pull request by odysa**
Add serialize for `TableMetadata` and use `TryFrom` as discussed in #177

### CODE REVIEW

**Review by liurenjie1024 — APPROVED** [review]
LGTM, thanks!

**Comment by liurenjie1024** [comment]
cc @ZENOTME @Xuanwo @Fokko PTAL

File: crates/iceberg/src/spec/table_metadata.rs [hunk 1]
```diff
@@ -676,9 +690,10 @@ pub(super) mod _serde {
         }
     }
 
-    impl From<TableMetadata> for TableMetadataV1 {
-        fn from(v: TableMetadata) -> Self {
-            TableMetadataV1 {
+    impl TryFrom<TableMetadata> for TableMetadataV1 {
+        type Error = Error;
+        fn try_from(v: TableMetadata) -> Result<Self, Error> {
```
[the remark this answers is not in the document]
Comment (odysa) [reply crates/iceberg/src/spec/table_metadata.rs:695]: Well, cuz I want make 2 separate PRs -- 1 for `serialize` and 1 for removing `expect` . I should really mention it before.

**Comment by odysa** [comment]
Replaced `expect` with `ok_or` in this PR as suggested.  cc @liurenjie1024 @ZENOTME

**Review by liurenjie1024 — APPROVED** [review]
Thanks!

**Review by Fokko — APPROVED** [review]
Thanks @odysa for raising this PR, and thanks @liurenjie1024 for the prompt review 🙌

**Pull request merged**